components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/utils/RESTAPIAdminClient.java

```diff
@@ -102,21 +108,112 @@ public void updateApi(APITemplateBuilder builder, String tenantDomain) throws Ax
 	}
 
 	/**
-	 * Delete the WebApp from Gateway
+	 * Deletes versioned web app configuration from the gateway
+	 *
 	 * @param tenantDomain
 	 * @throws AxisFault
 	 */
-	public void deleteApi(String tenantDomain) throws AxisFault {
+	public void deleteVersionedWebApp(String tenantDomain) throws AxisFault {
 		try {
 			if (tenantDomain != null && !("").equals(tenantDomain) &&
-			    !tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+					!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
 				restApiAdminStub.deleteApiForTenant(qualifiedName, tenantDomain);
 			} else {
 				restApiAdminStub.deleteApi(qualifiedName);
 			}
-			
+
 		} catch (Exception e) {
 			throw new AxisFault("Error while deleting WebApp", e);
 		}
 	}
+
+	/**
+	 * Adds non-versioned web app configuration to the gateway
+	 *
+	 * @param builder
+	 * @param tenantDomain
+	 * @throws AxisFault
+	 */
+	public void addNonVersionedWebApp(APITemplateBuilder builder, String tenantDomain)
+			throws AxisFault {
+
+		try {
+			String apiConfig = builder.getConfigStringForNonVersionedWebAppTemplate();
+			if (tenantDomain != null && !("").equals(tenantDomain)
+					&& !tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+				restApiAdminStub.addApiForTenant(apiConfig, tenantDomain);
+			} else {
+				restApiAdminStub.addApiFromString(apiConfig);
+			}
+		} catch (Exception e) {
+			throw new AxisFault("Error publishing non-versioned web app to the gateway", e);
+		}
+	}
+
+	/**
+	 * Updates non-versioned web app configuration in the gateway
+	 *
+	 * @param builder
+	 * @param tenantDomain
+	 * @throws AxisFault
+	 */
+	public void updateNonVersionedWebApp(APITemplateBuilder builder, String tenantDomain)
+			throws AxisFault {
+		try {
+			String apiConfig = builder.getConfigStringForNonVersionedWebAppTemplate();
+			if (tenantDomain != null && !("").equals(tenantDomain) &&
+					!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+
+				restApiAdminStub.updateApiForTenant(qualifiedNonVersionedWebAppName, apiConfig,
+													tenantDomain);
+			} else {
+				restApiAdminStub.updateApiFromString(qualifiedNonVersionedWebAppName, apiConfig);
+			}
+		} catch (Exception e) {
+			throw new AxisFault("Error while updating non-versioned web app in the gateway", e);
+		}
+	}
+
+	/**
+	 * Deletes non-versioned web app configuration form the gateway
```

Comment:
- Method comment: fixed in commit 818ca74c18433a3a2ffa7a6f80e8ef2b09fffa70 of PR #153
 - if-clause in [line 164](https://github.com/wso2/carbon-appmgt/pull/63/files#diff-2307f44ef8d71badbcf0995c22e643caR164): fixed in commit 50d5da1ffa1fffacf500b8005728d72302726392 of PR #153 
 - Catching `Exception`: fixed in commit d3987526bb0a0ed43659c34771aa8cf38c040af0 of PR #153